internal/configs/bootnode/config.go

```diff
@@ -0,0 +1,252 @@
+	ShardID   uint32          // ShardID of this node; TODO ek – revisit when resharding
+	role      Role            // Role of the node
+	Port      string          // Port of the node.
+	IP        string          // IP of the node.
+	RPCServer RPCServerConfig // RPC server port and ip
+	DebugMode bool            // log every single process and error to help to debug the syncing issues
+	NtpServer string
+
+	shardingSchedule shardingconfig.Schedule
+	networkType      NetworkType
+	DNSZone          string
+	WebHooks         struct {
+		Hooks *webhooks.Hooks
+	}
+	TraceEnable bool
+}
+
+// RPCServerConfig is the config for rpc listen addresses
+type RPCServerConfig struct {
+	HTTPEnabled bool
+	HTTPIp      string
+	HTTPPort    int
+
+	HTTPTimeoutRead  time.Duration
+	HTTPTimeoutWrite time.Duration
+	HTTPTimeoutIdle  time.Duration
+
+	WSEnabled bool
+	WSIp      string
+	WSPort    int
+
+	DebugEnabled bool
+
+	NetworkRPCsEnabled bool
+
+	RpcFilterFile string
+
+	RateLimiterEnabled bool
+	RequestsPerSecond  int
+}
+
+// configs is a list of node configuration.
+// It has at least one configuration.
+// The first one is the default, global node configuration
+var shardConfigs []ConfigType
+var defaultConfig ConfigType
+
+// GetDefaultConfig returns default config.
+func GetDefaultConfig() *ConfigType {
+	return &defaultConfig
+}
+
+// SetDefaultRole ..
+func SetDefaultRole(r Role) {
+	defaultConfig.role = r
+}
+
+func (conf *ConfigType) String() string {
+	return fmt.Sprintf("%v", conf.ShardID)
+}
+
+// SetShardID set the ShardID
+func (conf *ConfigType) SetShardID(s uint32) {
+	conf.ShardID = s
+}
+
+// SetRole set the role
+func (conf *ConfigType) SetRole(r Role) {
+	conf.role = r
+}
+
+// GetShardID returns the shardID.
+func (conf *ConfigType) GetShardID() uint32 {
+	return conf.ShardID
+}
+
+// Role returns the role
+func (conf *ConfigType) Role() Role {
+	return conf.role
+}
+
+// SetNetworkType set the networkType
+func SetNetworkType(networkType NetworkType) {
+	defaultConfig.networkType = networkType
+	for i := range shardConfigs {
+		shardConfigs[i].networkType = networkType
+	}
+}
+
+// GetNetworkType gets the networkType
+func (conf *ConfigType) GetNetworkType() NetworkType {
+	return conf.networkType
+}
+
+// SetVersion set the version of the node binary
+func SetVersion(ver string) {
+	version = ver
+}
+
+// GetVersion return the version of the node binary
+func GetVersion() string {
+	return version
+}
+
+// SetPeerID set the peer ID of the node
+func SetPeerID(pid peer.ID) {
+	peerID = pid
+}
+
+// GetPeerID returns the peer ID of the node
+func GetPeerID() peer.ID {
+	return peerID
+}
+
+// ShardingSchedule returns the sharding schedule for this node config.
+func (conf *ConfigType) ShardingSchedule() shardingconfig.Schedule {
+	return conf.shardingSchedule
+}
+
+// SetShardingSchedule sets the sharding schedule for this node config.
+func (conf *ConfigType) SetShardingSchedule(schedule shardingconfig.Schedule) {
```

Comment:
i don't think this is used by the bootnode right ?

Reply:
removed